core/src/main/java/hudson/model/Queue.java

```diff
@@ -760,17 +762,21 @@ public boolean cancel(Item item) {
     }
 
     /**
-     * Called from {@code queue.jelly} and {@code entries.jelly}.
+     * Called from {@code queue.jelly} and {@code queue-items.jelly}.
      */
     @RequirePOST
-    public HttpResponse doCancelItem(@QueryParameter long id) throws IOException, ServletException {
+    public HttpResponse doCancelItem(@QueryParameter long id) {
         Item item = getItem(id);
         if (item != null) {
             if(item.hasCancelPermission()){
-                cancel(item);
+                if(cancel(item)) {
+                    return HttpResponses.status(HttpServletResponse.SC_NO_CONTENT);
+                }
+                return HttpResponses.error(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Could not cancel run for id " + id);
             }
+            return HttpResponses.error(422, "Item for id (" + id + ") is not cancellable");
         } // else too late, ignore (JENKINS-14813)
-        return HttpResponses.forwardToPreviousPage();
+        return HttpResponses.error(422, "Provided id (" + id + ") does not exist");
```

Comment:
404 instead?

Reply:
Seconded it should be 404. if the id is provided the request is not malformed

Reply:
Fixed!